packages/flutter/lib/src/services/text_input.dart

```diff
@@ -805,17 +812,34 @@ RawFloatingCursorPoint _toTextPoint(FloatingCursorDragState state, Map<String, d
   return RawFloatingCursorPoint(offset: offset, state: state);
 }
 
-class _TextInputClientHandler {
-  _TextInputClientHandler() {
-    SystemChannels.textInput.setMethodCallHandler(_handleTextInputInvocation);
+/// A handler for the [SystemChannels.textInput] channel.
+///
+/// This class is used internally by the Flutter SDK and is only visible for
+/// testing.
+@visibleForTesting
+class TextInputClientHandler {
+  /// Creates a [TextInputClientHandler].
+  @visibleForTesting
+  TextInputClientHandler(this._channel, this._currentConnection) : assert(_channel != null) {
+    _channel.setMethodCallHandler(_handleTextInputInvocation);
   }
 
+  final MethodChannel _channel;
+
   TextInputConnection _currentConnection;
 
   Future<dynamic> _handleTextInputInvocation(MethodCall methodCall) async {
     if (_currentConnection == null)
       return;
     final String method = methodCall.method;
+
+    // Reattach request needs to be handled regardless of the client ID.
+    if (method == 'TextInputClient.reattach') {
```

Comment:
Same question here as the engine side PR. Can we call this something like `requestExistingInputState` or `requestInputConfiguration` so that we can avoid implying a previous attachment and express the goal of the message?

Reply:
Done